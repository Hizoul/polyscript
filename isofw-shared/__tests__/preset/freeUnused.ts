import freeUnused from "isofw-shared/src/tests/preset/freeUnused"

freeUnused()
