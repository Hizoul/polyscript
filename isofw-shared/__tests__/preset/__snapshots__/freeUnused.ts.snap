// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`free unused presets after program Update works as planned:  before aanything  1`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presets": false,
  },
  "doingQuery": Object {},
  "lists": Object {
    "presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
      ],
      "limit": 5,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "presetModel": 20,
  },
  "pageSize": 5,
  "previousQuery": Object {
    "presetModel": Object {
      "$limit": 5,
      "$skip": 0,
      "$sort": Object {
        "_id": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;

exports[`free unused presets after program Update works as planned: after reservation of four presets 1`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presetAssistant": true,
    "presets": false,
  },
  "doingQuery": Object {},
  "lists": Object {
    "presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "b19999999999999999999999",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "b19999999999999999999999",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "b19999999999999999999999",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "b19999999999999999999999",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
      ],
      "limit": 5,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "presetModel": 20,
  },
  "pageSize": 5,
  "previousQuery": Object {
    "presetModel": Object {
      "$limit": 5,
      "$skip": 0,
      "$sort": Object {
        "_id": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;

exports[`free unused presets after program Update works as planned: presets of camera 1 are all free again again 1`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presetAssistant": true,
    "presets": false,
    "projects": true,
  },
  "doingQuery": Object {},
  "lists": Object {
    "presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
      ],
      "limit": 5,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "presetModel": 20,
  },
  "pageSize": 5,
  "previousQuery": Object {
    "presetModel": Object {
      "$limit": 5,
      "$skip": 0,
      "$sort": Object {
        "_id": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;

exports[`free unused presets after program Update works as planned: two presets of camera 1 are now reserved 1`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presetAssistant": true,
    "presets": false,
    "projects": true,
  },
  "doingQuery": Object {},
  "lists": Object {
    "presetModel": Object {
      "data": Array [
        Object {
          "_id": "c2bbbbbbbbbbbbbbbb098765",
          "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c2bbbbbbbbbbbbbbbb198765",
          "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c2bbbbbbbbbbbbbbbb298765",
          "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c2bbbbbbbbbbbbbbbb398765",
          "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c2bbbbbbbbbbbbbbbb498765",
          "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
      ],
      "limit": 5,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "presetModel": 20,
  },
  "pageSize": 5,
  "previousQuery": Object {
    "presetModel": Object {
      "$limit": 5,
      "$skip": 0,
      "$sort": Object {
        "_id": 1,
        "number": 1,
      },
      "camera": "c2bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;
